File: January/02_Number_of_Ways_to_Split_Array/ShaFeiii.cpp
#include <bits/stdc++.h>
using namespace std;


// c++ Solution
class Solution {
public:
    int waysToSplitArray(vector<int>& nums) {
        int n = (int)nums.size();
        long long tot = 0, cur = 0;
        for (int i = 0; i < n; ++i)
            tot += nums[i];
        int splits = 0;
        for (int i = 0; i < n - 1; ++i) {
            tot -= nums[i];
            cur += nums[i];
            if (cur >= tot) splits++;
        }
        return splits;
    }
};

int main() {

    return 0;
}

File: January/03_Number_of_Ways_to_Split_Array/ShaFeiii.py
class Solution(object):
    def waysToSplitArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        tot = sum(nums)
        cur = 0
        splits = 0
        for i in range(0, len(nums) - 1):
            tot -= nums[i]
            cur += nums[i]
            if cur >= tot:
                splits += 1

        return splits